apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/client/FaucetButton.tsx

```diff
@@ -0,0 +1,145 @@
+
+function formatTime(seconds: number) {
+  const rtf = new Intl.RelativeTimeFormat("en", { numeric: "auto" });
+  const hours = Math.floor(seconds / 3600);
+  const minutes = Math.floor((seconds % 3600) / 60);
+  if (hours > 0) {
+    return rtf.format(+hours, "hour");
+  }
+
+  if (minutes > 0) {
+    return rtf.format(+minutes, "minute");
+  }
+
+  return rtf.format(+seconds, "second");
+}
+
+export function FaucetButton({
+  chain,
+  ttlSeconds,
+  amount,
+}: {
+  chain: ChainMetadata;
+  ttlSeconds: number;
+  amount: string;
+}) {
+  const address = useActiveAccount()?.address;
+  const chainId = chain.chainId;
+  const definedChain = defineChain(chainId);
+  const faucetWalletBalanceQuery = useWalletBalance({
+    address: THIRDWEB_ENGINE_FAUCET_WALLET,
+    chain: definedChain,
+    client: thirdwebClient,
+  });
+  const claimMutation = useMutation({
+    mutationFn: async () => {
+      const response = await fetch("/api/testnet-faucet/claim", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          chainId: chainId,
+          toAddress: address,
+          amount,
+        }),
+      });
+
+      if (!response.ok) {
+        const data = await response.json();
+        throw data.error;
+      }
+    },
+    onSuccess: () => {
+      toast.success(
+        `${amount} ${chain.nativeCurrency.symbol} sent successfully`,
+      );
+      router.refresh();
+    },
+    onError: () => {
+      toast.error(`Failed to claim ${amount} ${chain.nativeCurrency.symbol}`);
+    },
+  });
+  const router = useRouter();
+
+  const isFaucetEmpty =
+    faucetWalletBalanceQuery.data !== undefined &&
+    faucetWalletBalanceQuery.data.value < toUnits("1", 17);
+
+  // not eligible to claim right now
+  if (ttlSeconds > 0) {
+    return (
+      <Button variant="outline" className="w-full" disabled>
+        Your next claim is available {formatTime(ttlSeconds)}
+      </Button>
+    );
+  }
+
+  // checking faucet balance
+  if (faucetWalletBalanceQuery.isLoading) {
+    return (
+      <Button variant="secondary" className="w-full gap-2">
+        Checking Faucet Balance <Spinner className="size-3" />
+      </Button>
+    );
+  }
+
+  // faucet is empty
+  if (isFaucetEmpty) {
+    return (
+      <Button variant="outline" disabled className="w-full">
+        Faucet is empty right now, come back later
+      </Button>
+    );
+  }
+
+  if (!address) {
+    return (
+      <CustomConnectWallet
+        loginRequired={false}
+        connectButtonClassName="!w-full !rounded !bg-primary !text-primary-foreground !px-4 !py-2 !text-sm"
+      />
+    );
+  }
+
+  // eligible to claim and faucet has balance
+  return (
+    <div className="flex flex-col w-full text-center">
+      <Button
+        variant="primary"
+        className="w-full gap-2"
+        onClick={async () => {
+          claimMutation.mutate();
+        }}
```

Comment:
try this flow for the toast:

```ts
const claimMutationPromise = claimMuation.mutateAsync();
toast.promise(claimMutationPromise, {...})
```

we have other places where we use `toast.promise` and it feels really good imo, try it out

Reply:
nice - changed